client/src/App.tsx

```diff
@@ -0,0 +1,55 @@
+import React, { useState, ChangeEvent, FormEvent } from 'react';
+
+function App() {
+    // Update state to specify that file can be `File` or `null`
+    const [file, setFile] = useState<File | null>(null);
+
+    // Type the event as `ChangeEvent<HTMLInputElement>`
+    const handleFileChange = (event: ChangeEvent<HTMLInputElement>) => {
+        if (event.target.files) {
+            setFile(event.target.files[0]);  // Ensure file is not null
+        }
+    };
+
+    // Type the submit event as `FormEvent<HTMLFormElement>`
+    const handleSubmit = async (event: FormEvent<HTMLFormElement>) => {
+        event.preventDefault();
+
+        if (!file) {
+            alert('Please select a file first.');
+            return;
+        }
+
+        const formData = new FormData();
+        formData.append('file', file);
+
+        try {
+            const response = await fetch('https://localhost:8080/upload', {
+                method: 'POST',
+                body: formData,
+            });
+
+            if (response.ok) {
+                const message = await response.text();
+                alert(message);
+            } else {
+                alert('File upload failed.');
+            }
+        } catch (error) {
+            console.error('Error during file upload:', error);
+            alert('An error occurred.');
+        }
+    };
+
+    return (
+        <div>
```

Comment:
Subsequently, the TSX element needs to have the classes specified from styles, which can be achieved in the following way:

```js
return (
        <div className={styles.wrapper}>
            <h1 className={styles.title}>Upload a File</h1>
            <form onSubmit={handleSubmit}>
                <input type="file" onChange={handleFileChange} />
                <button className={styles.button} type="submit">Upload</button>
            </form>
        </div>
    );
```